src/node/share_manager.h

```diff
@@ -47,24 +47,37 @@ namespace ccf
       recovery_threshold(recovery_threshold_)
     {
       shares = shares_;
-      ccf::crypto::sharing::Share secret;
       ccf::crypto::sharing::recover_unauthenticated_secret(
         secret, shares, recovery_threshold);
       data = secret.key(KZ_KEY_SIZE);
     }
 
+    LedgerSecretWrappingKey(ccf::crypto::sharing::Share& secret_)
+    {
+      secret = secret_;
+      num_shares = 1;
+      shares.resize(num_shares);
+      shares.emplace_back(secret_);
+      data = secret_.key(KZ_KEY_SIZE);
+    }
+
     LedgerSecretWrappingKey(
       std::vector<SecretSharing::Share>&& shares_, size_t recovery_threshold_) :
       recovery_threshold(recovery_threshold_)
     {
-      auto secret = SecretSharing::combine(shares_, shares_.size());
-      data.resize(secret.size());
-      std::copy_n(secret.begin(), secret.size(), data.begin());
-      OPENSSL_cleanse(secret.data(), secret.size());
+      // TBD (gsinha): How to set ccf::crypto::sharing::Share secret member
+      // variable here?
```

Comment:
It's data (the raw secret) prefixed with an index 0, if we do that, we might as well drop data altogether and avoid keeping the secret twice.

Reply:
I wasn't sure how to instantiate a Share instance using the constructor inputs. If we can do that then the wrap/unwrap methods also would need to create a temporary secret.key(size) instance to replace the current data instance usage in those methods?

Reply:
There is no way to do that in the current API, it's the lines 153 to 162 of sharing.cpp, but it's deliberately not exposed and the sampling is combined with the sharing to avoid sharing of unsafe keys/unsafe sizes.

I think that is fine, because re-constructed keys never need to compute a zero share, and so all objects created with this ctor should never need this member attribute. This is why I was suggesting splitting this into two distinct types: SharedLedgerSecretWrappingKey and ReconstructedLedgerSecretWrappingKey.

Reply:
Ok, @achamayou  we can take this comment when we get to SharedLedgerSecretWrappingKey/ReconstructedLedgerSecretWrappingKey refactoring work?